### ISSUE TITLE
@std/yaml parses yaml with anchor wrong

### ISSUE THREAD

**Issue by taind772** [denoland/std#4792]
**Describe the bug**

```ts
import * as yaml from "jsr:@std/yaml@0.224.0"

const sample = `
test: &test
    a: 2
    b: 4
    c: 6
another:
    <<: *test
    c: 8
    d: 10
`
console.log(yaml.parse(sample))
```

**Steps to Reproduce**
```sh
❯ deno run yaml-test.ts
{
  test: { a: 2, b: 4, c: 6 },
  another: { "0": undefined, "1": undefined, "2": undefined, c: 8, d: 10 }
}
```

**Expected behavior**
```sh
❯ echo "
  test: &test
      a: 2
      b: 4
      c: 6
  another:
      <<: *test
      c: 8
      d: 10
  " | yq -o json
{
  "test": {
    "a": 2,
    "b": 4,
    "c": 6
  },
  "another": {
    "a": 2,
    "b": 4,
    "c": 8,
    "d": 10
  }
}
```

**Environment**
- OS: Linux / Arch
- deno version: 1.43.3
- std version: 0.224

**Comment by taind772** [comment]
I think the wrong part is merging.
Support or not, the output doesn't make sense to me at all. If merging is not supported, at least we should return something like
```
{
  test: { a: 2, b: 4, c: 6 },
  another: { "<<": { a: 2, b: 4, c: 6 }, c: 8, d: 10 }
}
```

I found a workaround (for my use case) by using https://www.npmjs.com/package/yaml instead:
```ts
import * as yaml from "npm:yaml"

const sample = `
test: &test
    a: 2
    b: 4
    c: 6
another:
    <<: *test
    c: 8
    d: 10
`
console.log(yaml.parse(sample, { merge: true }))
```

**Comment by babiabeo** [comment]
This issue has been fixed by #5185.
